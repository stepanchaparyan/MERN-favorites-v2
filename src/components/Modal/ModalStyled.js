import styled from 'styled-components';

export const customStyles = {
  content: {
    top: '20%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
    display: 'flex',
    flexDirection: 'column',
    padding: '0',
    minWidth: '350px',
    minHeight: '200px',
    backgroundColor: '#f8f8f8',
    position: 'absolute'
  },
  overlay: {
    position: 'absolute',
    top: 55,
    backgroundColor: 'rgba(255, 255, 255, 0.75)'
  }
};

export const ButtonConfirm = styled.button`
  border: 0;
  border-radius: 4px;
  background-color: ${props => props.theme.cadetblue};
  font-weight: 600;
  font-size: 16px;
  cursor: pointer;
  margin: 12px 0;
  padding: 8px;
  min-width: 120px;
`;

export const ButtonClose = styled.button`
  background-color: ${props => (props.bgColor ? props.bgColor : props.theme.cadetblue)};
  border: 0;
  font-weight: 600;
  font-size: 16px;
  border-radius: 4px;
  cursor: pointer;
  margin-right: 8px;
  outline: none;
`;

export const ButtonCancel = styled.button`
  width: 30%;
  border: 0;
  background-color: ${props => (props.color ? props.color : 'lightgrey')};
  font-weight: 600;
  font-size: 16px;
  border-radius: 4px;
  cursor: pointer;
  margin: 12px 0;
  padding: 8px;
`;

export const ModalContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: ${props => (props.isBigSize ? '450px' : 'auto')};
`;

export const ModalTitleContainer = styled.div`
  background-color: ${props => (props.bgColor ? props.bgColor : props.theme.cadetblue)};
  display: flex;
  justify-content: space-between;
`;

export const ModalTitle = styled.div`
  font-size: 20px;
  padding: 16px;
  font-weight: bold;
`;

export const ModalTextContainer = styled.div`
  display: flex;
  min-height: 80px;
  justify-content: center;
  color: ${props => props.theme.cadetblue};
  font-size: 18px;
  font-weight: bold;
  align-items: center;
  padding: 0 4%;
  text-align: center;
`;

export const ModalButtonsContainer = styled.div`
  display: flex;
  justify-content: space-between;
  border-top: 1px solid lightGray;
`;

export const FooterButtons = styled.div`
  display: flex;
  justify-content: 'space-between';
`;

export const FooterContainer = styled.div`
  width: 100%;
`;
