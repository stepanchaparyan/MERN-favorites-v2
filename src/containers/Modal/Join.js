import React from 'react';
import { Modal } from 'react-bootstrap';

function Join(props) {
  const { closeModal, userName, handleNameChange, isSubmitDone, joinChat } = props;

  return (
    <Modal show onHide={closeModal}>
      <Modal.Body>
        <div>
          <input type="text" placeholder="Your name" value={userName} onChange={handleNameChange} />
          {isSubmitDone && userName.length === 0 && <p>Name is required</p>}
        </div>
      </Modal.Body>
      <Modal.Footer>
        <div>
          <button onClick={joinChat}>Join</button>
        </div>
      </Modal.Footer>
    </Modal>
  );
}

export default Join;
