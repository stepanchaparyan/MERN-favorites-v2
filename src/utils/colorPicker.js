import theme from '../styles/theme';
import { FAVITEM } from '../constants';

const { red, blue, limeGreen, orange, black } = theme;
const { TYPE } = FAVITEM;

export const colorPicker = category => {
  let color;
  switch (category) {
    case TYPE.FILM:
      color = red;
      break;
    case TYPE.MUSIC:
      color = blue;
      break;
    case TYPE.BOOKS:
      color = limeGreen;
      break;
    case TYPE.OTHER:
      color = orange;
      break;
    default:
      color = black;
  }
  return color;
};
