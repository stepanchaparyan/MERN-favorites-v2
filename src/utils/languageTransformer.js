const dict = {
  // lang - country
  ar: 'sa',
  be: 'bl',
  cs: 'sz',
  da: 'dk',
  de: 'de',
  el: 'gr',
  en: 'us',
  es: 'es',
  fa: 'ir',
  fr: 'fr',
  ja: 'jp',
  hi: 'in',
  hu: 'hu',
  hr: 'hr',
  hy: 'am',
  it: 'it',
  kk: 'kz',
  nl: 'nl',
  no: 'no',
  pl: 'pl',
  pt: 'pt',
  ru: 'ru',
  sk: 'sk',
  sl: 'sl',
  sv: 'se',
  tr: 'tr',
  uk: 'ua'
};

export const languageTransformer = listList => {
  let newList = [];
  listList.map(lang => {
    const item = Object.keys(dict).find(item => item === lang);
    if (item) {
      newList.push(dict[item].toUpperCase());
    }
  });
  return newList;
};

function getKeyByValue(object, value) {
  return Object.keys(object).find(key => object[key] === value);
}

export const setLanguage = (code, setLocale) => {
  const countryCode = Object.values(dict).includes(code.toLowerCase());

  if (countryCode) {
    setLocale(getKeyByValue(dict, code.toLowerCase()));
  }
};
