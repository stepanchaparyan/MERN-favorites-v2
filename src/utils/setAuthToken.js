import axios from 'axios';

const Authorization = 'Authorization';

const setAuthToken = token => {
  if (token) {
    axios.defaults.headers.common = { Authorization: `Bearer ${token}` };
  } else {
    delete axios.defaults.headers.common[Authorization];
  }
};
export default setAuthToken;
