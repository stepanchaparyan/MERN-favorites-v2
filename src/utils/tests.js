import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { shallow } from 'enzyme';

export function renderToJSON(Component, props = {}, Router) {
  if (Router) {
    return renderer
      .create(
        <Router>
          <Component {...props} />
        </Router>
      )
      .toJSON();
  }
  return renderer.create(<Component {...props} />).toJSON();
}

export function renderSnapshot(Component, props = {}) {
  const wrapper = shallow(<Component {...props} />);
  expect(wrapper).toMatchSnapshot();
}
