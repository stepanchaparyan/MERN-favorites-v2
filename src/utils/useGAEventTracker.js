// eslint-disable-next-line no-unused-vars
import React from 'react';
import ReactGA from 'react-ga';

const useGAEventTracker = (category = 'Category') => {
  return (action = 'action', label = 'label') => {
    ReactGA.event({
      category,
      action,
      label,
    });
  };
};

export default useGAEventTracker;
