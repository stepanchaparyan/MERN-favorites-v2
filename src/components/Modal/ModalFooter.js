import React from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import { FooterContainer, ModalButtonsContainer, ButtonCancel, ButtonConfirm } from './ModalStyled';
import { NOOP } from '../../constants';

const ModalFooter = ({
  isPrimaryButtonDisabled,
  primaryButtonCallBack,
  primaryButtonMessage,
  secondaryButtonCallBack,
  secondaryButtonMessage,
  isSecondaryButtonDisabled,
}) => {
  const { formatMessage } = useIntl();

  return (
    <FooterContainer>
      <ModalButtonsContainer>
        {secondaryButtonMessage && (
          <ButtonCancel disabled={isSecondaryButtonDisabled} onClick={secondaryButtonCallBack}>
            {formatMessage(secondaryButtonMessage)}
          </ButtonCancel>
        )}
        {primaryButtonMessage && (
          <ButtonConfirm disabled={isPrimaryButtonDisabled} onClick={primaryButtonCallBack}>
            {formatMessage(primaryButtonMessage)}
          </ButtonConfirm>
        )}
      </ModalButtonsContainer>
    </FooterContainer>
  );
};

ModalFooter.propTypes = {
  isPrimaryButtonDisabled: PropTypes.bool,
  primaryButtonCallBack: PropTypes.func,
  primaryButtonMessage: PropTypes.object,
  secondaryButtonCallBack: PropTypes.func,
  secondaryButtonMessage: PropTypes.object,
  isSecondaryButtonDisabled: PropTypes.bool,
};

ModalFooter.defaultProps = {
  isPrimaryButtonDisabled: false,
  primaryButtonCallBack: NOOP,
  primaryButtonMessage: null,
  secondaryButtonCallBack: NOOP,
  secondaryButtonMessage: null,
  isSecondaryButtonDisabled: false,
};

export default ModalFooter;
