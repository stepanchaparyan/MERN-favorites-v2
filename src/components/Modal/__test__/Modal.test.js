import { renderSnapshot } from '../../../utils/tests';
import Modal from '../Modal.js';

const props = {
  modalIsOpen: true,
  closeModal: jest.fn(),
  onConfirm: jest.fn(),
  title: 'title',
  text: 'text',
  buttonConfirmText: 'text',
  buttonCancelText: 'text',
  titleBgColor: 'blue',
  cancelButtonColor: 'red'
};

describe('Modal component test with Enzyme', () => {
  test('renders correct snapshot', () => {
    renderSnapshot(Modal, { ...props });
  });
});
