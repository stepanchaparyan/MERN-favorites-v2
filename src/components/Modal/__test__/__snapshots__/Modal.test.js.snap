// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component test with Enzyme renders correct snapshot 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  contentElement={[Function]}
  contentLabel="Modal"
  isOpen={true}
  onRequestClose={[MockFunction]}
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "backgroundColor": "#f8f8f8",
        "bottom": "auto",
        "display": "flex",
        "flexDirection": "column",
        "left": "50%",
        "marginRight": "-50%",
        "minHeight": "200px",
        "minWidth": "350px",
        "padding": "0",
        "position": "absolute",
        "right": "auto",
        "top": "20%",
        "transform": "translate(-50%, -50%)",
      },
      "overlay": Object {
        "backgroundColor": "rgba(255, 255, 255, 0.75)",
        "position": "absolute",
        "top": 55,
      },
    }
  }
>
  <styled.div
    isBigSize={false}
  >
    <styled.div
      bgColor="blue"
    >
      <styled.div>
        title
      </styled.div>
      <styled.button
        bgColor="blue"
        onClick={[MockFunction]}
      >
        x
      </styled.button>
    </styled.div>
    <styled.div>
      text
    </styled.div>
    <styled.div>
      <styled.button
        color="red"
        onClick={[MockFunction]}
      >
        text
      </styled.button>
      <styled.button
        onClick={[MockFunction]}
      >
        text
      </styled.button>
    </styled.div>
  </styled.div>
</Modal>
`;
